R/utilities.R

```diff
@@ -281,6 +281,7 @@ get_abundance_sc_long <- function(.data, features = NULL, all = FALSE, exclude_z
   selected_features <- feature_df[(feature_df$feature %in% features), ]
   if (isTRUE(all)) {
     selected_features <- feature_df[feature_df$assay_id %in% assays_to_use, ]
+    warning("To avoid mixing features from different experiments only features from the Main experiment will be used!")
```

Comment:
for the long format, we can output multiple assays/experiments, as we have many abundance columns.

Anything I am missing?

Reply:
Initially, I was quite concerned about mixing assays from different experiments. It is prone to throwing weird errors for edge cases, e.g. where features have the same name in multiple assays in both wide and long format. I have refactored the code to account for most of these cases that I can think of but I think it'll need a bit more testing to rule out the odd scenario that breaks it.